### DISCUSSION TITLE
Failed to load 'plugin.py

### DISCUSSION THREAD

**Discussion by piacco** [addiejanssen/domoticz-solaredge-modbustcp-plugin#60] [Q&A]
Hi Addie,

I'have installed the plugin, but get an error, do you have a solution for this?

`Error: SolarEdge ModbusTCP: (SolarEdge_ModbusTCP) failed to load 'plugin.py', 
Python Path used was '/home/pi/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/:/usr/lib/python39.zip:/usr/lib/python3.9:/usr/lib/python3.9/lib-dynload:/usr/local/lib/python3.9/dist-packages:/usr/lib/python3/dist-packages:/usr/lib/python3.9/dist-packages'.

Error: SolarEdge ModbusTCP: Traceback (most recent call last):
Error: SolarEdge ModbusTCP: File "/home/pi/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/plugin.py", line 54, in <module>
Error: SolarEdge ModbusTCP: import solaredge_modbus
Error: SolarEdge ModbusTCP: File "/usr/local/lib/python3.9/dist-packages/solaredge_modbus/__init__.py", line 7, in <module>
Error: SolarEdge ModbusTCP: from pymodbus.client.sync import ModbusTcpClient
Error: SolarEdge ModbusTCP: ModuleNotFoundError: No module named 'pymodbus.client.sync'`

I'm using Domoticz version 2023.1 and the python version is: Python Version: 3.9.2

**Comment by addiejanssen** [comment]
It means that the plugin can not find one of the libraries that `solaredge_modbus` requires to communicate with the inverter.
Which version of `solaredge_modbus` did you install and how did you install it?

**Comment by piacco** [comment]
Hallo Addie,

Bedankt voor jou reactie.
Ik heb de plugin geïnstalleerd volgens de Domoticz wiki pagina.

#enter domoticz plugins directory
cd domoticz/plugins
#if git command is not installed, install it with apt install git
git clone https://github.com/addiejanssen/domoticz-solaredge-modbustcp-plugin.git
cd domoticz-solaredge-modbustcp-plugin
sudo pip3 install -r requirements.txt
#restart domoticz
sudo service domoticz.sh restart

Heb ook even op mijn andere raspberry de plugin geïnstalleerd en dat gaat probleemloos.
De rpi waar de plugin wel goed op draait daarop heb ik buster met Python versie 3.7.3 geïnstalleerd en op de andere rpi draait op bullseye met Python versie 3.9.2.
Op beide rpi’s heb ik Domoticz 2023.1 geïnstalleerd.

Heb met pip3 install solaredge_modbus de bibliotheek geïnstalleerd.
Kan het ook liggen aan de Python versie in combinatie met Domoticz?

Groeten,
Albert

Op 3 mrt. 2023 om 14:13 heeft Addie Janssen ***@***.***> het volgende geschreven:

﻿

It means that the plugin can not find the solaredge_modbus library which is required to communicate with the inverter.
Did you install that library as documented in the README file?

**Comment by Corvus71** [comment]
I just installed the plugin and also got the plugin.py error.
Could not find the new devices.
This fix (from: https://www.domoticz.com/forum/viewtopic.php?p=272142#p272142) worked for me

sudo pip3 uninstall pymodbus
sudo pip3 install pymodbus==2.4.0

Then restart domoticz ( sudo service domoticz.sh restart )
